pkg/fleet-manager/application/rollout_helper.go

```diff
@@ -397,14 +445,88 @@ func deleteResourceCreatedByKurator(ctx context.Context, namespaceName types.Nam
 	return nil
 }
 
+func deleteIngressCreatedByKurator(ctx context.Context, kubeClient client.Client, rollout *applicationapi.RolloutConfig) error {
+	ingress := &ingressv1.Ingress{}
+	namespaceName := types.NamespacedName{
+		Namespace: rollout.Workload.Namespace,
+		Name:      ingressName,
+	}
+	if err := kubeClient.Get(ctx, namespaceName, ingress); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to get ingress %s in %s", namespaceName.Name, namespaceName.Namespace)
+		}
+	} else {
+		// verify if the ingress were created by kurator
+		annotations := ingress.GetAnnotations()
+		if _, exist := annotations[RolloutIdentifier]; exist {
+			labels := ingress.GetLabels()
+			if set := sets.New(strings.Split(labels[ingressLabelKey], ",")...); set.Contains(rollout.ServiceName) {
+				if set.Len() == 1 {
+					if deleteErr := kubeClient.Delete(ctx, ingress); deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
+						return errors.Wrapf(deleteErr, "failed to Delete ingress %s in %s", namespaceName.Name, namespaceName.Namespace)
+					}
+				} else {
+					newRules := make([]ingressv1.IngressRule, 0)
+					for _, rule := range ingress.Spec.Rules {
+						if rule.Host != rollout.RolloutPolicy.TrafficRouting.Host {
+							newRules = append(newRules, rule)
+						}
+					}
+					ingress.Spec.Rules = newRules
+					labels[ingressLabelKey] = strings.Join(set.Delete(rollout.ServiceName).UnsortedList(), ",")
+					ingress.SetLabels(labels)
+					if err := kubeClient.Update(ctx, ingress); err != nil {
+						return errors.Wrapf(err, "failed to Update ingress %s in %s", namespaceName.Name, namespaceName.Namespace)
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// create/update ingress configuration
+func renderIngress(ingress *ingressv1.Ingress, rollout *applicationapi.RolloutConfig) {
```

Comment:
Is this only used by nginx?
Maybe the function name can be replaced to `renderNginxIngress`

Reply:
ok